### PR TITLE
Ipv6 support for CES

### PR DESCRIPTION
**Pull request by shubh1410**
Signed-off-by: Shubham Darokar (shubhamdarokar1410@gmail.com / shubham.dipak.darokar1@ibm.com)

This pull request adds following functionality to the toolkit:
- CES export IPs can now be IPv6 addresses as well. Earlier only IPv4 addresses were supported.

### CODE REVIEW

**Review by rajan-mis — COMMENTED** [review]
Please incorporate mentioned review comment.

File: roles/nfs/common/tasks/check.yml [hunk 1]
```diff
@@ -118,3 +118,16 @@ Please define the CES shared root file system mount point in the inventory."
...
    - ccr_status.rc == 0
    fail_msg: "CCR is not enabled on the cluster. CCR must be enabled to install protocols"
+
+- name: check|Check if interface is defined
+  assert:
+   that:
+   - "{{ scale_protocols.interface }}|length > 0"
+  when: scale_protocols.use_ipv6
+
+# requires netaddr python library(pip3 install netaddr)
```
[the remark this answers is not in the document]
Comment (shubh1410) [reply roles/nfs/common/tasks/check.yml:128]: Yes. Without this library, below task will throw an error

File: playbook.yml [hunk 2]
```diff
@@ -14,4 +14,4 @@
      - zimon/precheck
      - zimon/node
      - zimon/cluster
-     - zimon/postcheck
```
Comment (rajan-mis) [inline playbook.yml:17]: Ignore this file for checkin . don't do git add .  . run git add <your_modifed_file>

File: playbook_json_ces.yml [hunk 3]
```diff
@@ -7,19 +7,19 @@
...
-     - gui/node
-     - gui/cluster
-     - gui/postcheck
-     - zimon/precheck
-     - zimon/node
-     - zimon/cluster
-     - zimon/postcheck
-     - callhome/precheck
-     - callhome/cluster
+     #- core/precheck
```
Comment (rajan-mis) [inline playbook_json_ces.yml]: Same as above comment . before commit run , git diff to make sure your required changes is getting for merge.

File: vars/scale_clusterdefinition.json [hunk 4]
```diff
@@ -72,9 +72,11 @@
       "callhome_schedule": [daily,weekly]
   },
   "scale_protocols":{
-      "smb": false,
-      "nfs": false,
-      "export_ip_pool": [192.168.100.100,192.168.100.101],
+      "smb": true,
+      "nfs": true,
+      "use_ipv6": true,
+      "interface": [eth0,eth1],
```
Comment (rajan-mis) [inline vars/scale_clusterdefinition.json]: We should give example and `use_ipv6` , please change this flag to `scale_ces_ipv6_enable` . and it should be false

File: roles/nfs/common/tasks/configure.yml [hunk 5]
```diff
@@ -135,6 +135,18 @@
   with_items:
   - "{{ scale_protocols.export_ip_pool }}"
 
+- name: configure | Add IPv6 interface
+  shell:
+   cmd: "{{ scale_command_path }}mmces interface add --nic {{ item }}"
+  when: scale_protocols.scale_ces_ipv6_enable and scale_protocols.interface|length > 0
+  with_items:
+  - "{{ scale_protocols.interface }}"
```
Comment (rajan-mis) [inline roles/nfs/common/tasks/configure.yml:143]: I ran jenkins , its failing . Add when condition if its variabled defined . this check should done everywhere.

File: roles/nfs/common/tasks/check.yml [hunk 6]
```diff
@@ -118,3 +118,20 @@ Please define the CES shared root file system mount point in the inventory."
    that:
    - ccr_status.rc == 0
    fail_msg: "CCR is not enabled on the cluster. CCR must be enabled to install protocols"
+
+- name: Install python netaddr library
```
Comment (rajan-mis) [inline roles/nfs/common/tasks/check.yml:122]: We can't install through pip.  we need to use  yum, zypper or apt module based on os. sorry i forgot to add this in my review comment.

Comment (rajan-mis) [reply roles/nfs/common/tasks/check.yml:122]: Can we add when condition here as well , this should happen only if ipv6 is defined

File: roles/nfs/common/tasks/check.yml [hunk 7]
```diff
@@ -118,3 +120,29 @@ Please define the CES shared root file system mount point in the inventory."
    that:
    - ccr_status.rc == 0
    fail_msg: "CCR is not enabled on the cluster. CCR must be enabled to install protocols"
+
+- name: Prepare IPv6 export ip list
```
Comment (rajan-mis) [inline roles/nfs/common/tasks/check.yml:124]: Change header message to , check | 

For all others.

**Review by rajan-mis — APPROVED** [review]

**Pull request merged**